packages/application-generic/src/usecases/select-integration/select-integration.usecase.ts

```diff
@@ -52,6 +61,58 @@ export class SelectIntegration {
       return integrations[0];
     }
 
+    let integration: IntegrationEntity | null =
+      await this.getPrimaryIntegration(command);
```

Comment:
I'm concerned about getting the primary unnecessarily ... but lets get back to it